server/src/main/java/com/hexacore/tayo/reservation/ReservationService.java

```diff
@@ -0,0 +1,155 @@
+import com.hexacore.tayo.reservation.model.HostReservation;
+import com.hexacore.tayo.reservation.model.ReservationEntity;
+import com.hexacore.tayo.reservation.model.ReservationStatus;
+import com.hexacore.tayo.user.model.UserEntity;
+import jakarta.transaction.Transactional;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class ReservationService {
+
+    private final ReservationRepository reservationRepository;
+    private final CarRepository carRepository;
+    private final ImageRepository imageRepository;
+
+    @Transactional
+    public ResponseDto createReservation(CreateReservationDto createReservationDto) {
+        long guestUserId = 13L; // TODO: JWT 토큰에서 userId를 추출하고 로그인한 상태에서만 실행되도록 추가
+        // 현재 예시는 Guest 유저의 ID:13
+
+        // TODO: userRepository에서 userEntity 제공받기
+        UserEntity guestUser = UserEntity.builder().id(guestUserId).build();
+
+        CarEntity carEntity = carRepository.findById(createReservationDto.getCarId())
+                .orElseThrow(() -> new GeneralException(ErrorCode.CAR_NOT_FOUND));
+
+        UserEntity hostUser = carEntity.getOwner();
+
+        List<List<LocalDateTime>> possibleRentDates = carEntity.getDates();
+
+        LocalDateTime rentDate = createReservationDto.getRentDate();
+        LocalDateTime returnDate = createReservationDto.getReturnDate();
+
+        possibleRentDates.stream()
+                // date.get(0) ~ date.get(1) 사이의 날짜인지 검증
+                .filter(date -> date.get(0).isBefore(rentDate))
+                .filter(date -> date.get(1).isAfter(returnDate))
+                .findFirst()
+                .orElseThrow(() -> new GeneralException(ErrorCode.RESERVATION_DATE_NOT_IN_RANGE));
+
+        ReservationEntity reservationEntity = ReservationEntity.builder()
+                .guest(guestUser)
+                .host(hostUser)
+                .car(carEntity)
+                .rentDate(createReservationDto.getRentDate())
+                .returnDate(createReservationDto.getReturnDate())
+                .status(ReservationStatus.READY)
+                .build();
+        reservationRepository.save(reservationEntity);
+
+        return ResponseDto.success(HttpStatus.OK);
+    }
+
+    public ResponseDto getGuestReservations() {
+        // TODO: JWT 토큰에서 userId를 추출하고 로그인한 상태에서만 실행되도록 추가
+        long guestUserId = 13L;
+
+        List<ReservationEntity> reservationEntities = reservationRepository.findAllByGuest_id(guestUserId);
+        List<GuestReservation> guestReservations = new ArrayList<>();
+
+        for (ReservationEntity reservationEntity : reservationEntities) {
+            CarEntity carEntity = reservationEntity.getCar();
+            List<ImageEntity> imageEntities = imageRepository.findByCar_Id(carEntity.getId());
```

Comment:
이 부분도 양방향 매핑 되어 있어서 그냥 
```suggestion
            List<ImageEntity> imageEntities = carEntity.getImages();
```
이렇게 하면 될 거 같아요